### PR TITLE
Add users, teams, organisations to seeds and fix id generation

### PR DESCRIPTION
**Pull request by slawosz**
Generate organizations, teams and users in seeds. Also generate uuid's in database.

### CODE REVIEW

File: psd-web/db/migrate/20200225103936_generate_uuid_for_organizations_teams_and_users.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class GenerateUuidForOrganizationsTeamsAndUsers < ActiveRecord::Migration[5.2]
+  def change
+    safety_assured do
+      change_column_default(:organisations, :id, from: nil, to: "gen_random_uuid()")
```
Comment (slorek) [inline psd-web/db/migrate/20200225103936_generate_uuid_for_organizations_teams_and_users.rb]: 👏 good spot, we do need to generate IDs now we are no longer pulling from Keycloak.

File: psd-web/db/seeds.rb [hunk 2]
```diff
@@ -4,14 +4,15 @@
 def create_blob(filename, title: nil, description: nil)
   ActiveStorage::Blob.create_after_upload!(io: File.open("./db/seed_files/#{filename}"), filename: filename, content_type: "image/jpeg", metadata: {
     title: title || filename,
-    description: description,
-    updated: Time.now.iso8601
+      description: description,
+      updated: Time.now.iso8601
```
Comment (slorek) [inline psd-web/db/seeds.rb]: mis-aligned indentation

File: psd-web/db/seeds.rb [hunk 3]
```diff
@@ -408,9 +412,31 @@ def create_blob(filename, title: nil, description: nil)
 
   investigation.products << product
 
-  KeycloakService.sync_orgs_and_users_and_teams
+  # KeycloakService.sync_orgs_and_users_and_teams
```
Comment (slorek) [inline psd-web/db/seeds.rb]: ✂️

File: psd-web/db/seeds.rb [hunk 4]
```diff
@@ -408,9 +412,31 @@ def create_blob(filename, title: nil, description: nil)
...
+
+  organisation = Organisation.create!(name: "Office for Product Safety and Standards", path: "/Organisations/Office for Product Safety and Standards")
```
Comment (slorek) [inline psd-web/db/seeds.rb]: We can actually remove all of the `path` columns now as well.

File: psd-web/db/seeds.rb [hunk 5]
```diff
@@ -408,9 +412,31 @@ def create_blob(filename, title: nil, description: nil)
...
+
+  %i[opss_user psd_user user].each do |role|
+    UserRole.create!(user: user1, name: role)
+  end
+  %i[team_admin opss_user psd_user user].each do |role|
+    UserRole.create!(user: user2, name: role)
+  end
 
-  User.load_from_keycloak
+  # User.load_from_keycloak
```
Comment (slorek) [inline psd-web/db/seeds.rb]: ✂️

File: psd-web/db/seeds.rb [hunk 6]
```diff
@@ -408,9 +412,31 @@ def create_blob(filename, title: nil, description: nil)
...
-  KeycloakService.sync_orgs_and_users_and_teams
+  # KeycloakService.sync_orgs_and_users_and_teams
+
+  organisation = Organisation.create!(name: "Office for Product Safety and Standards", path: "/Organisations/Office for Product Safety and Standards")
+
+  enforcement = Team.create!(name: "OPSS Enforcement", path: "/Organisations/Office for Product Safety and Standards/OPSS Enforcement", team_recipient_email: "enforcement@example.com", "organisation": organisation)
+
+  Team.create!(name: "OPSS Science and Tech", path: "/Organisations/Office for Product Safety and Standards/OPSS Science and Tech", team_recipient_email: nil, "organisation": organisation)
+  Team.create!(name: "OPSS Trading Standards Co-ordination", path: "/Organisations/Office for Product Safety and Standards/OPSS Trading Standards Co-ordination", team_recipient_email: nil, "organisation": organisation)
+  Team.create!(name: "OPSS Incident Management", path: "/Organisations/Office for Product Safety and Standards/OPSS Incident Management", team_recipient_email: nil, "organisation": organisation)
```
Comment (slorek) [inline psd-web/db/seeds.rb]: I'm not sure if we need to create these teams if they're not used.

File: psd-web/db/migrate/20200225103936_generate_uuid_for_organizations_teams_and_users.rb (hunk 1 above)
Comment (slawosz) [reply psd-web/db/migrate/20200225103936_generate_uuid_for_organizations_teams_and_users.rb]: We do, please take a look into migrations in this PR.

File: psd-web/db/migrate/20200225103836_enable_pgcrypto.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+class EnablePgcrypto < ActiveRecord::Migration[5.2]
+  def change
+    safety_assured do
+      enable_extension "pgcrypto"
```
Comment (frankieroberto) [inline psd-web/db/migrate/20200225103836_enable_pgcrypto.rb]: Might be an idea to guard against the extension already being installed?

```suggestion
      enable_extension "pgcrypto" unless extension_enabled?("pgcrypto")
```

File: psd-web/db/schema.rb [hunk 8]
```diff
@@ -199,12 +200,10 @@
     t.index ["business_id"], name: "index_locations_on_business_id"
   end
 
-  create_table "organisations", id: :uuid, default: nil, force: :cascade do |t|
+  create_table "organisations", id: :uuid, default: -> { "gen_random_uuid()" }, force: :cascade do |t|
     t.datetime "created_at", null: false
     t.string "name"
-    t.string "path"
```
Comment (frankieroberto) [inline psd-web/db/schema.rb]: Looks like the "path" column has been removed from the schema, but I can't see a migration which has dropped this?

File: psd-web/.gitignore [hunk 9]
```diff
@@ -184,3 +184,7 @@ build-iPhoneSimulator/
 
 
 # End of https://www.gitignore.io/api/ruby,node,rails
+*.swp
+*.swo
+*.swn
+*.swm
```
Comment (frankieroberto) [inline psd-web/.gitignore]: Add a comment to explain what these are?

File: psd-web/db/seeds.rb [hunk 10]
```diff
@@ -260,6 +261,9 @@ def create_blob(filename, title: nil, description: nil)
     legal_name: "ABC limited"
   )
 
+  # migrations are causing model having wrong attributes,
+  # we need to reset them
+  Location.reset_column_information
```
Comment (frankieroberto) [inline psd-web/db/seeds.rb]: Bit confused by this... Is this just a problem with your local development environment?

File: psd-web/db/migrate/20200225103836_enable_pgcrypto.rb (hunk 7 above)
Comment (slawosz) [reply psd-web/db/migrate/20200225103836_enable_pgcrypto.rb]: There is no need: https://github.com/rails/rails/blob/f33d52c95217212cbacc8d5e44b5a8e3cdc6f5b3/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb#L386

File: psd-web/db/schema.rb (hunk 8 above)
Comment (slawosz) [reply psd-web/db/schema.rb]: Its here:)

File: psd-web/db/seeds.rb (hunk 10 above)
Comment (slawosz) [reply psd-web/db/seeds.rb]: It happens when you run migrate and seed in same process.

File: psd-web/db/schema.rb (hunk 8 above)
Comment (frankieroberto) [reply psd-web/db/schema.rb]: I swear this wasn't when I looked... ;-)

File: psd-web/db/migrate/20200225103836_enable_pgcrypto.rb (hunk 7 above)
Comment (frankieroberto) [reply psd-web/db/migrate/20200225103836_enable_pgcrypto.rb]: doh!

File: psd-web/.gitignore [hunk 11]
```diff
@@ -184,3 +184,8 @@ build-iPhoneSimulator/
 
 
 # End of https://www.gitignore.io/api/ruby,node,rails
+# Ignore vim temp files:
+*.swp
+*.swo
+*.swn
+*.swm
```
Comment (frankieroberto) [inline psd-web/.gitignore]: Add these to your local [`~/.gitignore_global` file](https://help.github.com/en/github/using-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer) instead? 😄

Comment (frankieroberto) [inline psd-web/.gitignore]: Add these to your local [`~/.gitignore_global` file](https://help.github.com/en/github/using-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer) instead? 😄

**Pull request merged**